iwyu.cc

```diff
@@ -1045,7 +1045,9 @@ class AstFlattenerVisitor : public BaseAstVisitor<AstFlattenerVisitor> {
     NodeSet* node_set = &nodeset_decl_cache_[decl];
     if (node_set->empty()) {
       if (decl->isImplicit()) {
-        TraverseImplicitDeclHelper(dyn_cast_or_null<FunctionDecl>(decl));
+        if (FunctionDecl* func = DynCastFrom(decl)) {
```

Comment:
As far as I could find, there really is nothing to traverse, since they are not defined in the source code. 
The BuiltinTemplateDecl contains only its ID and inherits TemplateDecl things, like parameters and so on.

Reply:
Makes sense! Might be nice to add a comment to say that.

Reply:
After experimenting for a while (trying to make it work for other decls) I went for the TODO comment.